packages/pages/src/scaffold/template/generate.ts

```diff
@@ -0,0 +1,117 @@
+import { ProjectStructure } from "../../common/src/project/structure.js";
+import path from "node:path";
+import fs from "node:fs";
+
+/* eslint-disable @typescript-eslint/no-unused-vars */
+// TODO: Remove after using adding generation code for templates
+
+export const generateTemplate = async (
+  projectStructure: ProjectStructure
+): Promise<void> => {
+  const questions: PromptObject[] = [
+    {
+      type: "text",
+      name: "templateName",
+      message: "What would you like to name your Template?",
+      validate: (templateName) =>
+        validateTemplateName(templateName, projectStructure) ||
+        "Please ensure the name provided isn't already used and is valid.",
+    },
+    {
+      type: "confirm",
+      name: "isVisualEditor",
+      message: "Is this a Visual Editor template?",
+      initial: true,
+    },
+    {
+      type: (prev) => (prev ? null : "toggle"),
+      name: "isDynamic",
+      message: "Is this a static or dynamic template?",
+      initial: true,
+      active: "Dynamic",
+      inactive: "Static",
+    },
+    {
+      type: (prev) => (prev ? "select" : null),
+      name: "entityScope",
+      message:
+        "How would you like you to define the entity scope for your template?",
+      choices: [
+        { title: "Entity Type", value: "entityTypes" },
+        { title: "Static", value: "static" },
+        { title: "Filter", value: "savedFilterIds" },
+        { title: "Entity Id", value: "entityIds" },
+      ],
+    },
+    {
+      type: (prev, values) =>
+        values.entityScope === "entityTypes" ? "list" : null,
+      name: "filter",
+      message: "Enter the entity type(s):",
+      initial: "",
+      separator: ",",
+    },
+    {
+      type: (prev, values) =>
+        values.entityScope === "savedFilterIds" ? "list" : null,
+      name: "filter",
+      message: "Enter the saved filter ID(s):",
+      initial: "",
+      separator: ",",
+    },
+    {
+      type: (prev, values) =>
+        values.entityScope === "entityIds" ? "list" : null,
+      name: "filter",
+      message: "Enter the entity ID(s):",
+      initial: "",
+      separator: ",",
+    },
+  ];
+
+  const response = await prompts(questions);
+
+  // TODO (SUMO-5251): handle generating VE templates
+  // TODO (SUMO-5252): handle generating non-VE templates
+};
+
+// Returns true if templateName can be formatted into valid filename and that filename isn't being used.
+const validateTemplateName = (
+  templateName: string,
+  projectStructure: ProjectStructure
+): boolean => {
+  const formattedFileName = formatFileName(templateName);
+
+  // Must start with an alphabetic char
+  if (/^[^a-zA-Z]/.test(formattedFileName)) {
+    return false;
+  }
+
+  const templatePath = path.join(
+    projectStructure.getTemplatePaths()[0].path,
+    formattedFileName
+  );
+  if (fs.existsSync(templatePath)) {
+    return false;
+  }
+
+  return true;
+};
+
+const formatFileName = (templateName: string): string => {
```

Comment:
Do we have this already too? If not, add docs for what it does.

Reply:
The code for it is in alpha unfortunately.